src/TensorTwoD.cpp

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2015 - 2022, Intel Corporation
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+
+#include "config.h"
+
+#include <cmath>
+#include <algorithm>
+#include <functional>
+#include <numeric>
+
+#include "TensorTwoD.hpp"
+
+#include "geopm/Exception.hpp"
+#include "TensorOneD.hpp"
+
+
+namespace geopm
+{
+
+
+    TensorTwoD::TensorTwoD(std::size_t rows, std::size_t cols)
+    {
+        set_dim(rows, cols);
+    }
+
+    TensorTwoD::TensorTwoD(const TensorTwoD &other)
+    {
+        m_mat = other.m_mat; 
```

Comment:
Leverage std::vector's initializer (considering TensorOneD provides a copy constructor and assignment operator):

TensorTwoD(const TensorTwoD &other)
    : m_mat(other.m_mat)